netket/graph/lattice.py

```diff
@@ -531,13 +493,130 @@ def position_from_basis_coords(self, basis_coords: CoordT) -> PositionT:
+            Valid wave vector components in this basis are integers in (periodic BCs)
+            or zero (in open BCs).
+
+        Throws an `InvalidWaveVectorError` if any of the supplied wave vectors
+        are not reciprocal lattice vectors of the simulation box.
+        """
+        # Ensure that ks has at least 2 dimensions
+        ks = _np.asarray(ks)
+        if ks.ndim == 1:
+            ks = ks[_np.newaxis, :]
+
+        result = ks @ self._lattice_dims.T / (2 * pi)
+        # Check that these are integers
+        is_valid = is_approx_int(result)
+        if not _np.all(is_valid):
+            raise InvalidWaveVectorError(
+                "Some wave vectors are not reciprocal lattice vectors of the simulation box spanned by\n"
+                + "\n".join(
+                    [
+                        str(self._lattice_dims[i])
+                        + (" (PBC)" if self.pbc[i] else " (OBC)")
+                        for i in range(self.ndim)
+                    ]
+                )
+            )
+
+        result = _np.asarray(_np.rint(result), dtype=int)
+        # For axes with non-periodic BCs, the k-component must be 0
+        is_valid = _np.logical_or(self.pbc, result == 0)
+        if not _np.all(is_valid):
+            raise InvalidWaveVectorError(
+                "Some wave vectors are inconisistent with open boundary conditions"
+            )
+
+        return result
+
+    # Generating space groups
+    # -----------------------------------------------------------------------
+    def space_group_builder(
+        self, point_group: Optional[PointGroup] = None
+    ) -> "SpaceGroupBuilder":
+        """
+        Returns a `SpaceGroupBuilder` object that represents the spatial symmetries of `self`.
+
+        Arguments:
+            point_group: a `PointGroup` object describing the point-group
+                         symmetries of `self`. Optional, if not supplied, the
+                         `PointGroup` object provded at construction is used.
+
+        Returns:
+            A `SpaceGroupBuilder` object that generates `PermutationGroup`s
+            encoding the action of `point_group`, the translation group of `self`,
+            and the space group obtained as their semidirect product as
+            permutations of the sites of `self`. It also yields space group irreps
+            for symmetrising wave functions.
+        """
+        from .space_group import SpaceGroupBuilder
+
+        if point_group is None:
+            if isinstance(self._point_group, PointGroup):
+                point_group = self._point_group
+            elif isinstance(self._point_group, Callable):
+                self._point_group = self._point_group()
+                point_group = self._point_group
+            else:
+                raise TypeError(
+                    "space_group_builder() missing required argument 'point_group'\n(lattice has no default point group)"
+                )
```

Comment:
```suggestion
                raise RuntimeError(
                    "space_group_builder() missing required argument 'point_group'\n(lattice has no default point group)"
                )
```

Reply:
`TypeError`s are common for complaining about missing arguments

Reply:
Will you please go and fix the tests this breaks @femtobit.

Reply:
Hm, the default point group not being available felt like a different case for me. (I commited it quickly because I had it in my first commit for this function, I just forgot to include it.) But in any case, either is fine for me.

Reply:
I don't care, I'm just fed up with cleaning up tests, so you do that if you want to change stuff like this.

Reply:
Fair point.

In any case, I've fixed the tests and the unrelated formatting issue.